src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -538,7 +539,16 @@ where
         source.update()?;
     }
 
-    let deps = source.query_vec(&dep)?;
+    let deps = loop {
+        match source.query_vec(&dep)? {
+            Poll::Ready(deps) => {
+                break deps;
+            }
+            Poll::Pending => {
+                // TODO: dont hot loop for it to be Ready
```

Comment:
I forget the context in which this function is called, but this could call the hypothetical new `Source::block_on_stuff` method